### ISSUE TITLE
PyQt6: DLL load failed while importing QtWidgets: The specified procedure could not be found.

### ISSUE THREAD

**Issue by GravityWolfNotAmused** [marcelotduarte/cx_Freeze#2011]
**Prerequisite**
 - Make sure no duplicated issue has already been reported. You should look for closed issues, too.
   - #1685 mentioned the error but was stated to be fixed. I doubt the cause is the same reason. 
 - Make sure you are not asking us to help to solve your specific issue. GitHub issues are opened mainly for development purposes. If you want to ask someone to help to solve your problem, go to some community site like StackOverflow, etc.
   - Used Sample to keep it from specific solves. 
 - Make sure your problem is not derived from packaging (e.g. Homebrew).
   - **Not sure how to do this.**

**Describe the bug**
After building my app, and the sample application into executables, I receive the error below.

**To Reproduce**
- Verify that [for x64 (64 bit) Windows](https://aka.ms/vs/16/release/vc_redist.x64.exe) is installed on the server
- Grab the sample PyQt6
- Add include_msvcr to build_exe options
- Build the project with ```python setup.py build```
- Upload to teams
- Download build folder onto host system
- Run EXE
- Observe Error

**Expected behavior**
I expect the sample and my application to open when ran from the executable.

**Screenshots**
![image](https://github.com/marcelotduarte/cx_Freeze/assets/24533882/45adcbf6-7e24-471d-acad-233d8b87c051)

**Desktop (please complete the following information):**
 - Platform information (e.g. Ubuntu Linux 22.04): Build on Windows 11, Ran on Windows Server 2012 Azure VM
 - OS architecture (e.g. amd64): Both are amd64
 - cx_Freeze version [e.g. 6.11]: 6.15.5
 - Python version [e.g. 3.10]: Build on: Python 3.9.10, Ran with no system install of Python.
 - PyQt version: 6.5.2

**Additional context**
The end goal of all of this is to understand what needs to be done to package my application, and/or the sample application without the need for the Windows Server to install Python.

I originally tried to build and run my application; however, due to trying to make sure that this isn't specific to my app, I tried the sample project that was packaged with the repo. This issue sounds like #1685 on the surface; however, there was no user resolution to this issue.

**Comment by marcelotduarte** [comment]
I suspect that I should include this in the docs:
If you are using an older Windows version than Windows 10 and the latest system updates are not installed, [Universal C Runtime](https://support.microsoft.com/en-us/help/2999226/update-for-universal-c-runtime-in-windows) might be also required.

Can you confirm?

**Comment by GravityWolfNotAmused** [comment]
I have installed the provided Universal C Runtime. Afterward, the error still exists as originally posted. Both the sample project and my project have the same error on the system that uses Windows Server 2012.

As mentioned, I build on Windows 11 and run on Windows Server 2012 without Python installed.

```python
"""A simple script to demonstrate PyQt6."""

from __future__ import annotations

import sys

from PyQt6.QtCore import Qt
from PyQt6.QtWidgets import QApplication, QLabel, QWidget


def main():
    app = QApplication(sys.argv)
    window = QWidget()
    window.setWindowTitle("Simple")
    window.setGeometry(300, 300, 300, 300)
    label = QLabel(window)
    label.setText("Hello World!")
    label.setGeometry(0, 0, 300, 300)
    label.setAlignment(Qt.AlignmentFlag.AlignCenter)
    window.show()
    return app.exec()


if __name__ == "__main__":
    sys.exit(main())
```

```python
"""
A simple setup script to create an executable using PyQt6. This also
demonstrates how to use excludes to get minimal package size.

test_pyqt6.py is a very simple type of PyQt6 application.

Run the build process by running the command 'python setup.py build'

If everything works well you should find a subdirectory in the build
subdirectory that contains the files needed to run the application.
"""

from __future__ import annotations

import sys

from cx_Freeze import Executable, setup

try:
    from cx_Freeze.hooks import get_qt_plugins_paths
except ImportError:
    get_qt_plugins_paths = None

include_files = []
if get_qt_plugins_paths:
    # Inclusion of extra plugins (since cx_Freeze 6.8b2)
    # cx_Freeze imports automatically the following plugins depending of the
    # use of some modules:
    # imageformats, platforms, platformthemes, styles - QtGui
    # mediaservice - QtMultimedia
    # printsupport - QtPrintSupport
    for plugin_name in (
        # "accessible",
        # "iconengines",
        # "platforminputcontexts",
        # "xcbglintegrations",
        # "egldeviceintegrations",
        "wayland-decoration-client",
        "wayland-graphics-integration-client",
        # "wayland-graphics-integration-server",
        "wayland-shell-integration",
    ):
        include_files += get_qt_plugins_paths("PyQt6", plugin_name)

# base="Win32GUI" should be used only for Windows GUI app
base = "Win32GUI" if sys.platform == "win32" else None

build_exe_options = {
    "bin_excludes": ["libqpdf.so", "libqpdf.dylib"],
    # exclude packages that are not really needed
    "excludes": ["tkinter", "unittest", "email", "http", "xml", "pydoc"],
    "include_files": include_files,
    "zip_include_packages": ["PyQt6"],
    "include_msvcr": True
}

executables = [Executable("main.py", base=base)]

setup(
    name="simple_pyqt6",
    version="0.1",
    description="Sample cx_Freeze PyQt6 script",
    options={"build_exe": build_exe_options},
    executables=executables,
)
```

Out of honesty and feeling, I do feel like there is just something I am missing here as nothing I have come across has been helpful or led me to any different errors or advancement in solving this issue. I once again appreciate your time.

**Comment by GravityWolfNotAmused** [comment]
For my Azure VM configuration:
![image](https://github.com/marcelotduarte/cx_Freeze/assets/24533882/9f0b3eb8-2124-425c-bd74-c2887d0b7f16)

**Comment by marcelotduarte** [comment]
We have to make some considerations: your app depends on pyqt6 and cx_Freeze and both depend on python. The python you use, [Python 3.9](https://www.python.org/downloads/release/python-390/) only works on windows 8 and above, so dependencies apply. cx_Freeze runs wherever python runs. As for pyqt6, you have to check if it runs on windows 2012, which is very old. I have no way to test, among the machines I can access and test, it has 2016 onwards.

**Comment by marcelotduarte** [comment]
I think this is related: https://stackoverflow.com/questions/74380258/qt-qpa-plugin-could-not-load-the-qt-platform-plugin-windows-even-though-it-was

**Comment by GravityWolfNotAmused** [comment]
So, from it seems there are a few take aways from this conversation.

Firstly, try an older version of python as the version I'm using is not ment for Windows Server 2012.

Secondly, down grade the the latest version of PyQt that is compatible with Python 3.8.

I'll work on things things and report back with my findings, I assume that there will be others that are trying to build utilities on legacy operating systems.

I appreciate your time, and we will speak soon.

**Comment by GravityWolfNotAmused** [comment]
I no longer have any issues with packaging my application or the sample application with cx_freeze. I solved my issue by downgrading from Python 3.9 to Python 3.8, which was the latest version supported by Windows Server 2012. I also downgraded my PyQt version from 6 to 5.

After both of these things were done and the proper dependencies were installed, my EXE ran flawlessly.

Anyone reading this in the future, be mindful of the latest version compatibility with your UI libraries, Python, and your OS.
